packages/smarthr-ui/src/components/Table/ThCheckbox.tsx

```diff
@@ -1,48 +1,73 @@
 import React, { forwardRef, useMemo } from 'react'
 import { tv } from 'tailwind-variants'
 
+import { Balloon } from '../Balloon'
 import { CheckBox, Props as CheckBoxProps } from '../CheckBox'
 import { Center } from '../Layout'
-import { VisuallyHiddenText } from '../VisuallyHiddenText'
 
 import { Th } from './Th'
 
 import type { DecoratorsType } from '../../types'
 
 type Props = {
-  decorators?: DecoratorsType<'checkAllInvisibleLabel'>
+  decorators?: DecoratorsType<'checkAllInvisibleLabel'> & {
+    checkColumnName?: (text: string) => string
+  }
 }
 
-const CHECK_ALL_INVISIBLE_LABEL = 'すべての行を選択'
+const CHECK_ALL_INVISIBLE_LABEL = 'すべての項目を選択/解除'
+const CHECK_COLUMN_NAME = '選択'
 
 const thCheckbox = tv({
   slots: {
-    inner: 'shr-absolute shr-inset-0 [&:not(:has([disabled]))]:shr-cursor-pointer',
+    inner: 'shr-absolute shr-inset-0 [&:not(:has([disabled]))]:shr-cursor-pointer shr-group/label',
     wrapper: 'shr-relative shr-w-[theme(fontSize.base)] [&]:shr-px-0.75',
     checkbox: '[&]:shr-block',
+    balloon: [
+      // 位置はセルの真ん中(50%)+checkboxの幅の半分(8px)+outlineの幅(4px)+Balloonの矢印の幅(5px), sr-onlyで隠す
+      'shr-absolute shr-left-[calc(50%+(theme(fontSize.base)/2)+4px+5px)] shr-sr-only',
+      // labelの中の要素に hover or focus-visible がある時のスタイル。shr-absoluteはshr-not-sr-onlyのpositionをabsoluteに上書きしている
+      'group-has-[:hover,:focus-visible]/label:shr-not-sr-only group-has-[:hover,:focus-visible]/label:shr-absolute group-has-[:hover,:focus-visible]/label:shr-whitespace-nowrap',
+    ],
   },
 })
 
 export const ThCheckbox = forwardRef<HTMLInputElement, CheckBoxProps & Props>(
   ({ decorators, className, ...others }, ref) => {
-    const { wrapperStyle, innerStyle, checkboxStyle } = useMemo(() => {
-      const { wrapper, inner, checkbox } = thCheckbox()
+    const { wrapperStyle, innerStyle, balloonStyle, checkboxStyle } = useMemo(() => {
+      const { wrapper, inner, balloon, checkbox } = thCheckbox()
       return {
         wrapperStyle: wrapper({ className }),
         innerStyle: inner(),
         checkboxStyle: checkbox(),
+        balloonStyle: balloon(),
       }
     }, [className])
 
+    const checkAllInvisibleLabel = useMemo(() => {
+      if (decorators && decorators.checkAllInvisibleLabel) {
+        return decorators.checkAllInvisibleLabel(CHECK_ALL_INVISIBLE_LABEL)
+      }
+
+      return CHECK_ALL_INVISIBLE_LABEL
+    }, [decorators])
```

Comment:
imo: こんな感じで書けそう（lint の関係で書けなかったらごめんなさい）

```suggestion
    const checkAllInvisibleLabel = useMemo(
      () => decorators?.checkAllInvisibleLabel?.(CHECK_ALL_INVISIBLE_LABEL) || CHECK_ALL_INVISIBLE_LABEL,
      [decorators]
    )
```

Reply:
AtsuhiMが言う通り、判定多いので、その辺分かりやすい方が良いかなって思ったので今の判定で行きます！

Reply:
smarthr-ui 全体で見ると、decorator がそもそも分かりにくくて “翻訳のお決まりのやつ” 感があるので、他にならってもいいかも?
FilterDropdown や SortDropdown は多くて参考になりそうでした。
https://github.com/kufu/smarthr-ui/blob/34eee9bb4f5048c7dfdd9c7ad29117d9ad066804/packages/smarthr-ui/src/components/Dropdown/FilterDropdown/FilterDropdown.tsx#L84C37-L103C4
https://github.com/kufu/smarthr-ui/blob/34eee9bb4f5048c7dfdd9c7ad29117d9ad066804/packages/smarthr-ui/src/components/Dropdown/SortDropdown/useSortDropdown.ts#L25-L42

Reply:
@uknmr @masuP9 
個人的には↑レベルでdecoratosが多いと更新判定のdecoratorsのチェックも馬鹿にならんのでuseMemoを一つにまとめるのもありかなぁ〜と思います